File: scripts/delivery-profile.py
#!/usr/bin/env python3
"""Read-only origin delivery diagnostics for the v0.9.27 deployment, not a benchmark."""
import argparse,datetime,json,os,re,subprocess,sys
from pathlib import Path
HOST='root@securityops.co'
SSH=['ssh','-p','5119','-o','ConnectTimeout=15',HOST]
COMMAND='docker exec -i --user apache --workdir /var/www/html/4get security-search php'
HEADER_KEYS={'content-type','content-encoding','cache-control','vary','server-timing','x-securitysearch-render'}
def clean_report(raw):
 if len(raw)>32768:raise RuntimeError('Report too large.')
 report=json.loads(raw)
 if not isinstance(report,dict) or type(report.get('schema'))is not int or report['schema']!=1:raise RuntimeError('Invalid report schema.')
 rows=report.get('samples')
 if not isinstance(rows,list)or len(rows)!=3:raise RuntimeError('Expected three fixed homepage observations.')
 out=[]
 for i,row in enumerate(rows,1):
  if not isinstance(row,dict)or row.get('sample')!=i or row.get('status')not in ('ok','unavailable'):raise RuntimeError('Invalid sample identity.')
  item={'sample':i,'status':row['status']}
  for key in ['http_status','curl_errno','wire_body_bytes','decoded_body_bytes']:
   n=row.get(key)
   if type(n)is not int or not 0<=n<=131072:raise RuntimeError('Invalid numeric sample.')
   item[key]=n
  for key in ['ttfb_ms','total_ms']:
   n=row.get(key)
   if type(n)not in (float,int)or not 0<=n<=10000:raise RuntimeError('Invalid timing.')
   item[key]=n
  asset=row.get('asset_version')
  if asset is not None and (type(asset)is not int or not 0<=asset<=9999):raise RuntimeError('Invalid asset version.')
  item['asset_version']=asset
  if item['status']=='ok' and (asset!=31 or item['http_status']!=200 or item['curl_errno']!=0):raise RuntimeError('Invalid success claim.')
  headers=row.get('headers')
  if not isinstance(headers,dict):raise RuntimeError('Invalid headers.')
  item['headers']={}
  for key in HEADER_KEYS:
   if key in headers:
    val=headers[key]
    if not isinstance(val,str)or len(val)>300 or re.search(r'[\x00-\x1f\x7f]',val):raise RuntimeError('Unsafe header field.')
    item['headers'][key]=val
  out.append(item)
 return {'schema':1,'scope':'Read-only localhost homepage HTTP inside the application container. Not public network/TLS/NPM or search-result performance.','samples':out}
def main():
 p=argparse.ArgumentParser(description=__doc__);p.parse_args()
 code=Path(__file__).with_suffix('.php')
 if not code.is_file()or code.is_symlink():raise RuntimeError('Keep the paired PHP probe with this script.')
 result=subprocess.run(SSH+[COMMAND],input=code.read_text(),text=True,capture_output=True,timeout=30)
 if result.returncode not in (0,2):raise RuntimeError('SSH/probe failed. No service changes were requested; inspect SSH access and the running container.')
 report=clean_report(result.stdout);report['observed_at_utc']=datetime.datetime.now(datetime.timezone.utc).isoformat();report['remote_probe_exit']=result.returncode
 folder=Path.home()/'Downloads';folder.mkdir(parents=True,exist_ok=True)
 stamp=datetime.datetime.now(datetime.timezone.utc).strftime('%Y%m%dT%H%M%S.%fZ')
 target=folder/('securitysearch-delivery-profile-'+stamp+'.json')
 fd=os.open(target,os.O_WRONLY|os.O_CREAT|os.O_EXCL,0o600)
 with os.fdopen(fd,'w')as f:json.dump(report,f,indent=2);f.write('\n')
 print(json.dumps(report,indent=2));print('Saved privately to '+str(target))
 return 0 if result.returncode==0 and all(row['status']=='ok'for row in report['samples']) else 2
if __name__=='__main__':
 try:sys.exit(main())
 except (Exception,KeyboardInterrupt)as error:
  print('Stopped: '+(str(error)if isinstance(error,RuntimeError)else type(error).__name__)+'. No container restart or configuration write was requested.',file=sys.stderr);sys.exit(1)

File: scripts/delivery-profile.php
<?php
/** Read-only, fixed localhost delivery observations; run only through the operator CLI. */
if (PHP_SAPI !== 'cli') { http_response_code(404); exit; }
if (!function_exists('curl_init')) { fwrite(STDERR,"PHP cURL required.\n"); exit(2); }
$rows=[];
for ($i=0;$i<3;$i++) {
    $body='';$headers=[];$handle=curl_init('http://127.0.0.1/');
    curl_setopt_array($handle,[CURLOPT_PROXY=>'',CURLOPT_FOLLOWLOCATION=>false,CURLOPT_PROTOCOLS=>CURLPROTO_HTTP,
        CURLOPT_CONNECTTIMEOUT_MS=>1000,CURLOPT_TIMEOUT_MS=>5000,CURLOPT_ENCODING=>'gzip',
        CURLOPT_HTTPHEADER=>['Host: securityops.co','User-Agent: SecuritySearch-DeliveryProfile/1.0','Connection: close'],
        CURLOPT_HEADERFUNCTION=>static function($ch,$line) use(&$headers){
            if (strlen($line)<=2048 && str_contains($line,':')) {
                [$key,$value]=explode(':',$line,2);$key=strtolower(trim($key));
                if(in_array($key,['content-type','content-encoding','cache-control','vary','server-timing','x-securitysearch-render'],true))
                    $headers[$key]=substr(preg_replace('/[\x00-\x1f\x7f]/',' ',trim($value)),0,300);
            } return strlen($line);
        },
        CURLOPT_WRITEFUNCTION=>static function($ch,$chunk) use(&$body){
            if(strlen($body)+strlen($chunk)>131072)return 0;$body.=$chunk;return strlen($chunk);
        }]);
    $ok=curl_exec($handle);$info=curl_getinfo($handle);$errno=curl_errno($handle);curl_close($handle);
    $match=[];$asset=null;
    if(preg_match('~/banner/securitysearch\.webp\?v([0-9]{1,4})~',$body,$match))$asset=(int)$match[1];
    $static=($headers['x-securitysearch-render']??'')==='static-home';
    $rows[]=['sample'=>$i+1,'status'=>$ok!==false && ($info['http_code']??0)===200 && $asset===33 && $static?'ok':'unavailable',
        'http_status'=>(int)($info['http_code']??0),'curl_errno'=>$errno,'asset_version'=>$asset,
        'ttfb_ms'=>round(($info['starttransfer_time']??0)*1000,3),'total_ms'=>round(($info['total_time']??0)*1000,3),
        'wire_body_bytes'=>(int)($info['size_download']??0),'decoded_body_bytes'=>strlen($body),'headers'=>$headers];
}
// No bodies, request cookies, process environment or upstream search requests.
echo json_encode(['schema'=>1,'scope'=>'Three fresh HTTP localhost connections inside the application container; no DNS/TLS/NPM/public-network or provider-search comparison.','samples'=>$rows],JSON_THROW_ON_ERROR|JSON_PRETTY_PRINT)."\n";
exit(count(array_filter($rows,static fn($row)=>$row['status']==='ok'))===3?0:2);
